docs/source/configuration/validation.md

```diff
@@ -0,0 +1,273 @@
+
+The most common thing is to have a field that you want to validate with a specific validator.
+Volto provide some default validators, see at the end of this chapter for more information.
+
+#### For Volto custom forms and block schema forms
+
+When you define them programatically, in your core, using JSON schema, you can register a custom validator using the `format` property.
+This is the case of creating the schema for a block:
+
+```ts
+let blockSchema = {
+  // ... fieldset definition in here
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'My custom URL field',
+      description: '',
+      format: 'url'
+    },
+  },
+  required: [],
+};
+```
+
+`url` named validator should be registered by this name as a Volto validator utility:
+
+```ts
+config.registerUtility({
+  type: 'validator',
+  name: 'url',
+  method: urlValidator,
+})
+```
+
+In this case, the `urlValidator` method validator will be applied for the block field `customField`.
+
+#### For content types
+
+Content types also can specify the `format` using the schema hints in the backend in the `frontendOptions`:
+
+```python
+from plone.supermodel import model
+from zope import schema
+
+class IMyContent(model.Schema):
+    directives.widget(
+        "customField",
+        frontendOptions={
+            "format": "url",
+        },
+    )
+    customField = schema.TextLine(
+        title="Custom URL field",
+        required=False,
+    )
+    # Rest of your content type definition
+```
+
+For the record, the resultant `plone.restapi` response will be something like the following, a bit different than in blocks JSON schema.
+But the validation engine will take care too:
+
+```json
+{
+  "properties": {
+    "customField": {
+      "title": "Custom URL field",
+      "widgetOptions": {
+        "frontendOptions": {
+          "format": "url"
+        }
+      }
+    }
+  }
+}
+```
+
+and the `urlValidator` method validator will be applied for the content type field `customField`.
+
+### Advanced scenarios
+
+In case you need more granularity and you don't have access to modify the existing implementation of the JSON schema definitions for existing content types, blocks or forms (might be in third party add-ons), you can use the following advanced validator registrations, using `field`, `widget`, `behaviorName` or `blockType` validator registrations.
+
+#### Per field `type` validators
+
+These validators are applied depending on the specified `type` of the field in the JSON schema from content types, forms or blocks.
+The next example is for the use case of JSON schema defined in a block:
+
+```ts
+let blockSchema = {
+  // ... fieldset definition in here
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'My custom field',
+      description: '',
+      type: 'integer',
+      maximum: 30
+    },
+  },
+  required: [],
+};
+```
+
+```ts
+config.registerUtility({
+  type: 'validator',
+  name: 'maximum',
```

Comment:
This is indeed part of the validation spec of json schema (same for maxLength, minLength, maxItems, pattern etc). Should we make these validators customisable? I think it would make more sense to have these "build-in" as validators since the name already implies what it should do.

Reply:
What we have today as built in:
- minLength
- maxLenght
- minimum
- maximum
- uniqueItem

We could implement the rest, for sure.